agents/acu/acu_agent.py

```diff
@@ -924,21 +924,18 @@ def _run_specified_scan(self, session, times, azs, els, vas, ves, azflags, elfla
         self.log.info('No more points in the queue')
         current_az = self.data['broadcast']['Corrected_Azimuth']
         current_el = self.data['broadcast']['Corrected_Elevation']
-        az_attempt_number = 0
-        while az_attempt_number < 10:
-            while round(current_az - end_az, 1) != 0.:
-                self.log.info('Waiting to settle at azimuth position')
-                yield dsleep(0.1)
-                az_attempt_number += 1
-                current_az = self.data['broadcast']['Corrected_Azimuth']
+        while round(current_az - end_az, 1) != 0.:
+            print('still rounding az')
+            self.log.info('Waiting to settle at azimuth position')
+            yield dsleep(0.1)
+            current_az = self.data['broadcast']['Corrected_Azimuth']
         if not azonly:
-            el_attempt_number = 0
-            while el_attempt_number < 10:
-                while round(current_el - end_el, 1) != 0.:
-                    self.log.info('Waiting to settle at elevation position')
-                    yield dsleep(0.1)
-                    el_attempt_number += 1
-                    current_el = self.data['broadcast']['Corrected_Elevation']
+            while round(current_el - end_el, 1) != 0.:
+                print('still rounding el')
+                self.log.info('Waiting to settle at elevation position')
+                yield dsleep(0.1)
+                current_el = self.data['broadcast']['Corrected_Elevation']
+        print('rounding completed')
```

Comment:
Ok, so now these loops will run indefinitely, unless the "current" positions reach the "end" positions.

I think there is still a potential for this to hang; imagine the situation where someone sees the telescope reach the end of the program, and then while it is still settling (i.e. it is stuck in these loops), they run stop-and-clear.  Then the telescope is stopped, near-to-but-not-quite-at the final positions.

So I think you either need to check for a mode switch (did someone send it to Stop?) or to simply bail out after some max total wait time (which I think was what you were thinking in the previous version).

You could do something like:
```
    give_up_time = time.time() + 5
    while round(current_az - end_az, 1) != 0. and time.time() < give_up_time:
      ...
```

I see that, below,  you do run .stop() before exiting...